### DISCUSSION TITLE
利用 Dataview + Blue Topaz + Markdown 辅助增强 + 导入到 Obsidian 插件，实现对标注的汇总与回顾

### DISCUSSION THREAD

**Discussion by kchen0x** [Kenshin/simpread#3807] [服务]
> 简悦官方注：以下内容来自 [简悦社区](https://t.me/simpread) 用户 [Kun Chen](https://github.com/kchen0x)  的文章，主要讨论了如何将导入的内容更好的汇总与展示，此套方案的特点是不使用同步助手即可实现自动导入到 Obsidian。

# 描述

利用 Dataview + Blue Topaz + Markdown 辅助增强 + 导入到 Obsidian 插件，可以方便的统计和归纳你的知识。

![Dataview效果图](https://user-images.githubusercontent.com/9739009/165709658-8f14d4e7-6a72-4c6c-8dfd-4c8e3f14651c.png)

![效果总览图](https://user-images.githubusercontent.com/81074/165702381-087a8614-3a68-4b2f-81a7-c94e2e80cc98.jpeg)

# 准备材料

- 简悦扩展（高级版）

  - [Markdown 模板辅助增强](https://simpread.ksria.cn/plugins/details/HD9GmoatXd) 全新的 Markdown 辅助增强插件

  - [导入到 Obsidan](https://simpread.ksria.cn/plugins/details/1VQ19jCD8Z) 

- Obsidian v0.14+

  - 主题： [Blue Topaz](https://github.com/whyt-byte/Blue-Topaz_Obsidian-css) 为了和教程达到相同的效果建议使用的主题，该主题目前为 OB 社区最火的主题之一

  - 插件： [Local REST API](obsidian://show-plugin?id=obsidian-local-rest-api) 为Obsidian 提供本地的 Rest API 接口，简悦可以把笔记通过接口发送到 Obsidian 中

  - 插件： [DataView](obsidian://show-plugin?id=dataview) （可选）实现文章向数据库的转换，通过一些查询语法，快速展示文章列表或表格

  - 插件： [Style Settings](obsidian://show-plugin?id=obsidian-style-settings) （可选）搭配主题使用，可以对主题进行全方位细致的修改

# 配置流程

## 简悦端插件配置

简悦端插件需要先安装准备条件的两个简悦插件，安装完成之后，找一篇文章进入「阅读模式」，右下角如果出现图示选项则说明安装成功：

![插件选项预览](https://user-images.githubusercontent.com/81074/165704990-c22cb8c0-de49-4f47-aae3-fe490eeecfa0.png)

如果明明安装了插件，却看不到插件选项，大概率是因为遇到了 [这个 bug](https://github.com/Kenshin/simpread/discussions/2342) ，可以根据教程进行修复。

### 1. 配置 Markdown 模板辅助增强

先打开 Markdown 模板辅助增强选项：

![Markdown 模板辅助增强设置页](https://user-images.githubusercontent.com/81074/165705058-118472fb-a052-49ef-b1a5-8ea56c7a982a.png)

1. 反向链接的标题：建议直接使用 `{{id}}-{{title}}{{mode}}` 模板即可。

2. 标签的前缀：`#`

3. 标签的后缀：不填写
   综上得到的标签效果会是这样：`#tag1`、`#tag2`、`#tagF/tagS`

4. 稍后读的模板：

   ```javascript
   ---
   UID: {{timestamp}}
   title: "srAnnote@{{title}}"
   alias: [<% if ( unread.note && unread.title != unread.note ) { %>"srAnnote@{{note}}",<% } %>"srAnnote@{{title}}"]
   type: Simpread
   index: {{idx}}
   ---
   
   # {{title}}
   
   > [!timeline]+ 简介
   >> **元数据**
   >---
   > **原文**:: [{{title}}]({{url}})
   > **日期**:: [[{{create|yyyy_mm_dd}}]]
   > **标签**:: #SimpRead {{tags}}
   <% if ( unread.refs ) { -%>
   > **外部引用**:: {{refs}}  
   <% } -%>
   >> **摘要**
   >---
   > <% if ( unread.desc ) { -%>
   <%- unread.desc %>
   <% } -%>
   
   ## 笔记
   <% if ( unread.annotations.length > 0 ) { %>
   {{annotations}}
   <% } %>
   ```

5. 标注的模板：

   ```javascript
   <%
   let colors = [ '#B4D9FB', '#ffeb3b', '#a2e9f2', '#a1e0ff', '#a8ea68', '#ffb7da' ],
       color  = colors[ annote.color ];
   -%>
   > [!<%= (annote.note? "note":"abstract")%>]+ <mark style="background-color: <%= color %>">Highlight</mark> [🧷](<{{an_int_uri}}>)[🌐](<{{an_ext_uri}}>) {{an_tags}}
   {{{html_format|>|{{an_html}}}}}
   <% if (annote.note) { -%>
   > - 📝{{an_note}}
   <% } -%>
   ^sran-{{an_id}}
    
   
   ```

6. 预览窗口：根据模板对当前阅读的文章进行解析。

7. 测试：点击就可以在 6 中看到预览效果。（如果配置了导出到 Obsidian 插件，「测试」时也会同时导出一份到 Obsidian 中）

8. 复制：可以一键将测试结果复制到剪贴板，此时可以粘贴到 Obsidian 中看看效果如何。

9. 展开预览：点击可以展开和缩回预览区域，便于编辑。

确认模板效果无误后，记得【保存】。

### 2. 配置导出到 Obisidian 选项

如果我们测试模板满足需求，就可以配置自动导出解析结果到 Obsidian 里面了。此时，需要借助 Obsidian 里面的 [Local REST API](obsidian://show-plugin?id=obsidian-local-rest-api) 插件来完成导入的过程，我们先在 Obsidian 中配置该插件：

![Local REST API 配置](https://user-images.githubusercontent.com/81074/165705278-5beb5f9e-ee01-4d02-b3ae-9ee68d73b573.png)

不建议修改接口的端口，保持默认的 `27123`，打开非安全 HTTP 连接。并复制当前的 Token 备用。

回到网页的阅读模式中，接下来我们配置导入到 Obsidian 选项：

1.  常规设置：根据你自己的 Obsidian 文档目录结构决定，测试的时候可以不填。
2.  导入方式：选择方案 2【使用 Local REST API 方式导出】，接口地址和端口，Token 可以从 Obsidian 中拿到。
3.  文件名设定：根据自己的情况定义就好，建议保持使用 `{{id}}-{{title}}{{mode}}`。
4.  导入全文：开启【添加 Metadata】。
5.  导入标注：开启【Markdown 模板辅助增强插件】。

配置完成，记住两个快捷键：

- `bbb` 导入标注信息到 Obsidian
- `ob` 导入全文到 Obsidian

效果如下：

![](https://user-images.githubusercontent.com/81074/165706329-6bf3959a-dd40-4b07-9a0f-e1639de9b55b.png)

## Dataview 设置

随着我们阅读量的积累，Obsidian 里面存放了足量的阅读信息之后，我们可以考虑创建一个索引页来快速导航，此时可以使用 Dataview。

    ```dataview
	table index as 序号, 日期, 标签
	where type = "Simpread"
	sort index asc
    ```

![Dataview效果图](https://user-images.githubusercontent.com/9739009/165709658-8f14d4e7-6a72-4c6c-8dfd-4c8e3f14651c.png)

## Blue Topaz 设置

Blue Topaz 是一个国人制作的可高度定制化的 Obsidian 主题，可以在主题市场中找到，安装完成之后不需要特殊的配置，当然也建议可以将主题的字体调整为【霞鹜文楷】。

![image](https://user-images.githubusercontent.com/9739009/165709786-3cbe9302-29d2-4823-bd12-d730a214244e.png)

# 补充

> 以下内容来自简悦官方。

1. 上述方案中的简悦配置部分，可以直接 [看这里](https://zhuanlan.zhihu.com/p/504280066) 的教程，内含可以使用 [自动化辅助增强](https://simpread.ksria.cn/plugins/details/etyod6A5ZT) 插件，实现自动化方案。

2. 也可以查看 [Markdown 官方模板库](https://github.com/Kenshin/simpread/discussions/2153)。

关于简悦与 Obsidian 的更多联动方案，[请看这里](https://github.com/Kenshin/simpread/discussions?discussions_q=label%3Aobsidian)。

**Comment by Kenshin** [comment]
嗨，各位。

简悦官方针对新用户或不喜欢配置的用户，发布了一个新功能：**Obsidan 配置库**。

细节请看下面，几分钟就能配置成完成。

***

![img](https://imgs.zhubai.love/c626678ff6b64dd1ac9dfdae6cf6d188.png)

> 针对新用户的一步到位的 Obsdian 配置库。

## 🗂 配置库

[简悦 · 配置库](https://www.yuque.com/kenshin/simpread/ds8zk0) 是简悦官方推出的一套针对新用户的极简配置方案，方便新用户用最快的方式使用简悦的各种高级服务，配置库内置了常用的双链笔记用法，如：Notion、Obsidian、Logseq、Roam Research，同时包含了简悦在阅读模式上的一些常规插件：Live Editor、题图、Safari 阅读模式等。

## 💡 特点

用户仅需要下载对应的配置库，修改里面的 UID 为自己的高级账户 UID，即可实现简悦的各种高级操作。

## 💁‍♂️ 受众

新用户或者不想使用手动配置的用户。

## 📝 如何使用

1️⃣ 下载

2️⃣ 修改为自己的 UID

3️⃣ 将必要的文件夹复制粘贴到相应位置

✅ Done！

## 📚 Obsidian 配置库

## 📝 特点

1️⃣ 加入稍后读，自动将本地快照和 Markdown 导入到 Obsidian

2️⃣ 加入标注时，自动将标注内容导入本地快照和 Obsidian

3️⃣ 改变稍后读或标注的元数据时，自动将改动的内容保存到本地快照和 Obsidian

4️⃣ 在 Obsidian 中直接内置好了简悦的标注系统，可直接在 Obsidian 实现标注

## 📖 区别

![img](https://imgs.zhubai.love/f4a51d8e8fb4401281b07f82bf0bc7d1.png)

## 📘 极速版

🔗 [Github](https://github.com/Kenshin/simpread-configs/blob/main/obsidian@little/Getting%20Started.md) | [语雀](https://www.yuque.com/kenshin/simpread/xkuecp)

不需要同步助手，无任何配置，支持 Markdown 形式的快照，自动同步标注，适合 Obsidain + 简悦的尝鲜用户。

## 📙 基础版

🔗 [Github](https://github.com/Kenshin/simpread-configs/blob/main/obsidian@localrestapi/Getting%20Started.md) | [语雀](https://www.yuque.com/kenshin/simpread/cg33gh)

需要同步助手，几乎无任何配置，支持本地快照，自动同步标注，适合 Obsidain + 简悦的轻量级用户。

## 📗 高级版

🔗 [Github](https://github.com/Kenshin/simpread-configs/blob/main/obsidian@sync/Getting%20Started.md) | [语雀](https://www.yuque.com/kenshin/simpread/wq35mh)

需要同步助手，几乎无任何配置，支持本地快照，自动同步标注，可在 Obsidian 中标注，适合 Obsidain + 简悦的重度用户。

**Reply by the1stnora** [reply]
这个高级版如果想要在自己的ob库里显示，路径该怎么设置呢～

**Reply by Kenshin** [reply]
@the1stnora 

可以看下 https://www.yuque.com/kenshin/simpread/wq35mh#comment-24952725 的回复
